Bluetooth "Hesse" variant

peterjc opened the pull request:
Updated to cover Ergogen, KiCad routing etc, case files (with your help - single commits to avoid the bloated history with the excessively large FreeCAD STEP files), and draft build guide.

---

This is for #20, currently only the Ergogen work (including a reset switch footprint & Hesse logo footprint) and basic documentation.

KiCad work to follow, probably as just a few commits as the history is too noisy with floating point number wobble as soon as you rotate the board back (it never returns to the exact same position).

3D printed case work to follow after that. Again, probably just a few commits - and I may ask you to try importing, using automatic refining/simplifying, and re-exporting to see if the bloated STEP files I get from FreeCAD can be slimmed down.

peterjc committed:
Starting work on a bluetooth variant, GT36K.

peterjc committed:
Setup Ergogen folder

peterjc committed:
Copy of TC36K ergogen configuration

peterjc committed:
Switch to ceoloide/mcu_supermini_nrf52840 footprint

The 'incidence graph of the affine plane of order 3' wiring will use
the bipartite 21 vertex split as 12 physical columns (including thumbs)
and 9 scanning rows. Thus placeholder wiring net entries C1 to C12,
and R1 to R9.

peterjc committed:
Support both choc v1 & v2 switches

This fixes an oversight in the TC36K design settings.

peterjc committed:
Remove Tutte-Coxeter Graph silkscreen art (using different graph here)

peterjc committed:
Move Supermini NRF52840 up 9mm to edge of PCB

Verified pin coordinates versus TC36K PCB.

Note this footprint does not include the top two pins for the battery connection...

peterjc committed:
Gamma Omega 'Hesse' is a better name than placeholder 'GT36K'.

The underlying graph of the affine plane of order 3 is also known as the
Hesse Configuration <https://en.wikipedia.org/wiki/Hesse_configuration>.
So what we are using could reasonably be called the 'Hesse Configuration
Incidence Graph'. A visually pleasing drawing of this is tricky, but the
Hesse Configuration Graph would work as a silk screen illustration.

peterjc committed:
Link to the excellent Kris Cables page on the SuperMini NRF52840

peterjc committed:
There is no reset button on the SuperMini NRF52840

Also, I intend to mount the controller (with its mid-mounted USB-C)
face up to expose the LEDs and get the USB at almost the same height.

peterjc committed:
Notes on the different board sizes & case implications

peterjc committed:
Add battery cut-out, eg 3.7v 110mAh 301230

That is a commonly used size of 3mm thickness, and 12mm by 30mm footprint.
This placement aims to leave a keepout at the bottom of the controller for
the bluetooth signal, but above the case taper and structural cross beam.
Will still need to adust the top and or bottom case for this...

This location should leave enough space either side of the battery space
for traces similar to the layout used on the TC36K.

I am currently thinking a battery with the JST socket, and wiring a matching
JST connector with wires to the battery terminals at the top of the controller
(by the USB port), possibly via an on/off switch. Running thick traces to this
battery location may be too tricky with the switch wiring in the way?

peterjc committed:
Workaround footprint oddity with v2 choc support

peterjc committed:
Put the hotswaps on the back (controller will be face up on front)

peterjc committed:
Switch first draft wiring (based on sorted matrix)

peterjc committed:
Swap C10, C9, C8 (left) with R4, R3, R8 (right)

peterjc committed:
Swap R1 & C3 (top right, cosmetic)

peterjc committed:
Move battery slot 3mm down to match bottom edge of current void in case

Top case will need cross-bar moved up for the desired battery placement.

peterjc committed:
Add JST battery socket (will move battery slot over)

This is on the left, even though the RAW pin is on the right, because
that turned out to be easier to route the traces to with my draft wiring
layout.

peterjc committed:
Move battery slot over for neighbouring JST power connection

Also made the battery cutout a fraction wider.

peterjc committed:
Add low-profile molex pico ezmate footprint next to JST

peterjc committed:
Silk screen version of the Hesse Configuration (12 lines & 9 points)

peterjc committed:
Use Hesse Configuration Graph as silk screen art

peterjc committed:
Describe Hesse Configuration Graph on back (only)

peterjc committed:
Add date and version of tools used in corner of PCB

peterjc committed:
Add name and version in corner of PCB

peterjc committed:
Pre-wire the hotswap pads & direct solder holes (on back)

peterjc committed:
Pre-wire each column (on back)

peterjc committed:
Updating to KiCad v9.0.4

peterjc committed:
Bigger battery...

peterjc committed:
Correct comment about Raspberry Pi Pico dimensions

peterjc committed:
Document how using ergogen_helper.py copy-traces

peterjc committed:
Use KiCad default trace width of 0.2, not 0.25

peterjc committed:
Centre row wiring on pad

peterjc committed:
Vertically center col wiring on hole

peterjc committed:
Right angle reset switch 1825027-8

peterjc committed:
Use 3D model for reset switch 1825027-8

From https://www.digikey.co.uk/en/models/1632543

peterjc committed:
Adjust auto-tracing for ground-plane routing

peterjc committed:
Flip battery connections to right-hand side.

peterjc committed:
Move the reset button more centrally

This is tighter to the screw hole, less tucked in the case corner.

peterjc committed:
Drop Molex battery option, rotate JST, enlarge battery cutout

peterjc committed:
Update ceoloide footprints for keep-out support

peterjc committed:
Used ceoloide footprint for bluetooth antenna keep-out

peterjc committed:
Move C5 down & R4, R3, R8 up for more space at extra trio

This should make wiring for the Nice!Nano v2 easy, which places its
three extra pins slightly up-and-left compared to the SuperMiniNRF52840
aka ProMicro52840

peterjc committed:
Trim the USB cutout enough for a ground fill along edge

Not actually sure we need the cutout with the Nice!Nano v2 or the
SuperMiniNRF52840 aka ProMicro52840 controller.

peterjc committed:
Improve visual alignment of radio keepout with battery cutout

peterjc committed:
Drop the USB cutout (not needed as flush with base of controller board)

I actually have my Supermini nRF52840 in hand to confirm this.

peterjc committed:
Swap to Nice Nano footprint to do the tighter wiring

Can swap back or add the Supermini nRF52840 pins later (within the
Nice Nano radio keepout).

peterjc committed:
Route PCB, add ground plane (copy earlier routing)

This is currently perfectly aligned on the Ergogen output (no rotation
noise giving slight offsets), allowing easy updates:

$ cp output/pcbs/gamma-omega-hesse.kicad_pcb hesse/pcb/gamma-omega-hesse-routed-new.kicad_pcb && \
            /Applications/KiCad/KiCad.app/Contents/Frameworks/Python.framework/Versions/Current/bin/python3 \
            resources/kb_ergogen_helper/ergogen_helper.py copy-traces \
            hesse/pcb/gamma-omega-hesse-routed.kicad_pcb hesse/pcb/gamma-omega-hesse-routed-new.kicad_pcb  \
            && open hesse/pcb/gamma-omega-hesse-routed-new.kicad_pcb
./src/common/stdpbase.cpp(59): assert ""traits"" failed in Get(): create wxApp before calling this
05:27:37 PM: Debug: Adding duplicate image handler for 'PNG file'
05:27:37 PM: Debug: Adding duplicate image handler for 'JPEG file'
05:27:37 PM: Debug: Adding duplicate image handler for 'TIFF file'
05:27:37 PM: Debug: Adding duplicate image handler for 'GIF file'
05:27:37 PM: Debug: Adding duplicate image handler for 'PNM file'
05:27:37 PM: Debug: Adding duplicate image handler for 'PCX file'
05:27:37 PM: Debug: Adding duplicate image handler for 'IFF file'
05:27:37 PM: Debug: Adding duplicate image handler for 'Windows icon file'
05:27:37 PM: Debug: Adding duplicate image handler for 'Windows cursor file'
05:27:37 PM: Debug: Adding duplicate image handler for 'Windows animated cursor file'
05:27:37 PM: Debug: Adding duplicate image handler for 'TGA file'
05:27:37 PM: Debug: Adding duplicate image handler for 'XPM file'
WARN: Skipped 290 existing traces
Copied 921 / 1211 traces.

Then manually removing the clashing from the battery routing,
and copying/redifining the ground zone.

peterjc committed:
Add Gamma Omega logo on silk screen in radio keepout

peterjc committed:
Use PCM_marbastlib-xp-promicroish:nice_nano_AH_USBup footprint

peterjc committed:
Polish routing at nice nano footprint; mark Supermini nRF52840 hole trio

peterjc committed:
Add cutout to to reach Supermini nRF52840 hole trio (hackery)

peterjc committed:
Update Hesse Ergogen README

peterjc committed:
Update Hesse main README

peterjc committed:
First draft of Hesse build guide

peterjc committed:
Parts details

peterjc committed:
Link to hesse-nice_nano_v2-zmk.uf2

peterjc committed:
Fix URL markdown

peterjc committed:
typos

peterjc committed:
Remove duplicated automated traces from Ergogen

peterjc committed:
Move Hesse reset switch 1mm in (for case clearance)

peterjc committed:
Top case for 'Hesse' Bluetooth variant (joint effort)

Based on the original top case with the following changes:

* Moved rib beams for new battery compartment.
* Moved beam for new reset button.
* New hole for reset button.
* Head room for reset switch.
* LED funnels in Hesse top-case (0.9mm thick skin) inspired by
  https://github.com/carrefinho/visorbearer case (even thinner).
* Raised the USB port slightly due to different controller and
  controller mounting approach.

Done in a mixture of tools, initially editing the original case in
FreeCAD (peterjc) which made the file much larger. USB cutout
adjusted and re-exported as a smaller STEP file by unspecworks.

peterjc committed:
Link Hesse case step files to controller footprint for viz

unspecworks commented:
Yeah, of course. I’m on vacation right now, so I’ll take a look after the trip this weekend.

peterjc commented:
Thanks - that'd be appreciated.

unspecworks commented:
For the Gamma Omega case, everything was designed with fairly large tolerances. Resin prints are naturally more precise, but JLC still lists their tolerance around 0.2 mm. So I think setting it to about 0.2 mm is probably the safer choice!

As for the LED, I just remembered I had a print done with JLC’s Black resin at 0.6 mm. It’s the bottom case from the TOTEM, and the hole in the center of the switch is 0.6 mm. Looks like that should be plenty! The blue LED looks about half as bright as the red one. maybe because of its shorter wavelength, or maybe just because of power draw limits. (And blue are not turns on state on photo)


![fkgerg](https://github.com/user-attachments/assets/0608ab44-d300-4322-a87c-5f7486054b7e)


I’ll check the USB fit as well.

unspecworks commented:
I used is infused-kim’s Nice!Nano v2 3D Model.


<img width="1467" height="1033" alt="Screenshot 2025-09-29 at 22 48 46" src="https://github.com/user-attachments/assets/e645fce9-6aa9-42cb-aa9d-2d668dda38b6" />



The USB cutout looks fine. It’s shifted a bit downward, but the tolerance seems generous enough. Still, since the case is meant for Hesse specifically, maybe moving it slightly upward would be a good idea.

---

<img width="840" height="726" alt="Screenshot 2025-09-29 at 22 50 13" src="https://github.com/user-attachments/assets/9ba4e88a-44a7-4e10-a44e-b36bdcc23477" />


The LED also looks fine. Based on the Nice!Nano v2, its position seems a little off, but on my Supermini nRF52840 it actually lines up. I don’t think it’ll matter much anyway since the guide shape should take care of it.

peterjc commented:
That’s all really encouraging - thanks again! I’ll update the 3D files …

peterjc committed:
Show infused-kim's Nice Nano v2 3D model in KiCad

Note offest and rotation, probably as I swapped out their Ergogen footprint
for this more detailed one.

unspecworks commented:
I’m not sure if you already mentioned this, but the MCU footprint looks like a Nice!Nano v2. Doesn’t that mean the extra pins aren’t compatible with clones like the SuperMini nRF52840?

peterjc commented:
Indeed, the PCB is designed to fit the Nice!Nano v2 perfectly, but thanks to the cut out at the end of the chip (in the radio keepout region) should work with the clones like the SuperMini nRF52840 too - with a slightly longer bit of soldering to reach their offset trio of central pins.
<img width="563" height="805" alt="Screenshot 2025-09-24 at 23 25 15" src="https://github.com/user-attachments/assets/9424e88c-786a-4a00-b6e4-e14f2d4b6b03" />

The faux-castellation plan putting the controller flat on the top of the PCB face up means some slightly trickier soldering anyway (I plan to use the provided hearer pins or diode legs). I think this will work...

peterjc commented:
Right now it is an unproven cunning plan (with the fallbacks being buying the original Nice!Nano v2, or tweaking the PCB design to move the holes and having another batch made), but I am optimistic 🤞

peterjc committed:
Use KiCad's 3D model for JST PH 2.0 battery connector

peterjc commented:
Touch wood that's the top case good to go too:
* Enlarged the reset hole
* Recreated the USB hole higher up. The curved surface didn't turn out the same as yours, an angle on a pocket opertation may not have been the right approach. Still, this is functional and at first glance the same.

If you could work your magic to shrink the STEP files and then view the composite once more, _hopefully_ we're good to place an order...

unspecworks commented:
<img width="1178" height="844" alt="Screenshot 2025-10-07 at 00 09 51" src="https://github.com/user-attachments/assets/304ebeb3-bbbe-4e41-9c7e-a6f467cdcf1b" />


I’ve checked a few things. As you mentioned, the USB cutout definitely needs some adjustment. If you have any trouble with it, I can handle the cutout revision myself.  just let me know.

<img width="774" height="888" alt="Screenshot 2025-10-07 at 00 14 27" src="https://github.com/user-attachments/assets/14651798-c09c-4215-b52b-0fe583ae102e" />
<img width="776" height="754" alt="Screenshot 2025-10-07 at 00 13 52" src="https://github.com/user-attachments/assets/39e06ef8-7c92-4fca-9fd0-20c51662f7af" />

Also, I think it’d be good to ensure there’s enough clearance between the battery connector and the upper case wall. (maybe  makes issue with leg solder pads)
Other than that, everything else looks fine to me.

<img width="1115" height="730" alt="Screenshot 2025-10-07 at 00 19 30" src="https://github.com/user-attachments/assets/e18832ed-87e4-487e-aa97-a94c1c846b0a" />

peterjc commented:
I've reverted the USB case change, and will gladly let you handle that. Thank you for the kind offer.

The second battery connector is not (yet) on the PR branch, it was only on the more experimental case branch. It's a lower profile Molex PicoBlade series connector 53261-0271, specifically a horizontal SMD one right now. The idea was to cater to batteries sold with the (made up) "JST 1.25mm" aka "Micro JST 1.25mm" labelled connector. It fits, but is probably not worth bothering with. If you did like the idea, then a through hole version would be more secure and easier to solder, but means another round of case edits. I'll just remove it (git commit reverted on that branch).

unspecworks committed:
Bottom case for 'Hesse' Bluetooth variant (joint effort)

Based on the original covered bottom case with the following changes:

* Simlified the hotswap cutouts no covered (lighter this way).
* Added 1.4mm battery resess cutout
* Added cutout for the JST PH 2.0 battery connector legs
* Added cutout for the reset button legs
* Added cutout for the Nice!Nano v2 or clone middle pins

This retains the full length header cutouts for the Raspberry Pi Pico,
so should still work with the original or TC36K PCB too.

Done in a mixture of tools, initially editing the original case in
FreeCAD (peterjc) which made the file much larger. Imported back
into AF, glitches cleaned up, and re-exported as a smaller STEP file
(unspecworks).

peterjc committed:
Fix top case offset for KiCad 3D model usage

peterjc committed:
Assign LCSC parts for hotswap, JST battery, and reset switch

peterjc committed:
Hesse Gerbers ZIP, BOM & positions etc from KiCad Fabrication-Toolkit

See https://github.com/bennymeg/Fabrication-Toolkit

Positions not validated - unable to pre-order the hotswap sockets
and so JLCPLC would not let me proceed with an assembly order.

unspecworks commented:
Great.  merged.
If your prototype works properly, you can go ahead and finalize the remaining documentation.
Thanks for your contribution!

peterjc commented:
Case and PCB arrived, and after a quick mock assembly the flashing bright blue LED indicting no battery (on the clone controller, this would be a red LED on the real Nice!Nano v2) is visible as a nice muted level. Not easy to catch on camera, it seemed brighter than this photo suggests (will take more with a solid LED once assembled and charging, and have the user LED configured):

![IMG_3483](https://github.com/user-attachments/assets/7dddc89b-944d-466a-8049-406a3d2b3c19)

The cut out for bridging the three pins from their Nice!Nano v2 location to that on the clone controller looks doable but fiddly.

I won’t be able to try soldering all this immediately (equipment issues).

The bottom case which was ordered in the discounted random resin is white, with some slight unwanted texture but fine for the underneath.

peterjc commented:
Good news, the red LED is also visible through the black case - but muted compared to the blue.

![Blue and red LEDs](https://github.com/user-attachments/assets/8b10db67-e0e3-4b30-8c66-88900af1fe61)

Note worthy, the hairpin bent wiring for the three middle pins needs to be as flush as possible as the bottom case spacing there in minimal. It might be worth making that cut out a little deaper.

Good news, the reset button works nicely.

Good news, I worked out how to do the faux-castellation smoothly - still starting with opposite corners: Break off a block of three header pins at once, move the block to one end, and insert the pins from below through the PCB and controller, solder on the top (controller side). Repeat. Remove a block of three (leave rest in place for stability), solder the pins, repeat. Can then trim, reflow, and tackle the three middle pins.

Bad news, something isn't right once the controller is in place - same thing seems to have happened again on my second PCB and spare controller. I will seek suggestions for diagnosis on the ZMK Discord...

peterjc commented:
Third time lucky - this time one of the red clones rather than black, but I think the first two failures were probably due to me using too much solder on the controller pins. Not quite finished, but the controller and 36 hotswaps seem to be working.

I used a modified method for the faux-castellation which I will add to the build guide. Break the header pins into blocks of three or four header (to make removal easier). Push the pins further throught the blocks leaving about 2mm at the short end. Solder these into the PCB with the block and long ends facing up, solder on the back (same side as the hotswap pads). Carefully thread the controller onto the pins, perhaps at an angle to help ensure a connection. Plug in the USB and test shorting each key's switch. The right thumbs and two right-most colums can be tested using the three middle pins. Remove the controller, remove the header blocks. Fix any pins lost. Replace the controller flat on the PCB this time. Clip the pins, and solder them using as little solder as possible. Retest. Do the three middle pins as before. Retest.
